Common/src/linear_solvers_structure.cpp

```diff
@@ -463,7 +476,7 @@ unsigned long CSysSolve::FGMRES_LinSolver(const CSysVector & b, CSysVector & x,
   if (monitoring) {
```

Comment:
@economon, I noticed that here you did not add the FULL_COMMS condition, do you want me to?

Reply:
If you have the time, sure. That would be great.

The framework is not complete yet, but as I am sure you have figured, the idea is to have COMM_NONE, COMM_MINIMAL, COMM_FULL. COMM_FULL is default, COMM_MINIMAL reduces the MPI comms slightly by disabling things like residual reductions during runtime that do not affect correctness, and COMM_NONE will eventually disable all comms during runtime (purely for "ideal" performance and scalability benchmarking, which will obviously be incorrect in terms of the solution).

Reply:
There is quite a lot of work on low communication Krylov methods, things like not reducing dot products for example, I am not using enough cores to make that an issue yet (I think) but it could be a nice feature.
I'll just make FGMRES consistent with CG and BCGSTAB.